src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -345,8 +427,26 @@ public Task InvokeWithCancellationAsync(string targetName, IReadOnlyList<object>
         /// <exception cref="ObjectDisposedException">If this instance of <see cref="JsonRpc"/> has been disposed.</exception>
         public Task<Result> InvokeWithCancellationAsync<Result>(string targetName, IReadOnlyList<object> arguments = null, CancellationToken cancellationToken = default(CancellationToken))
         {
-            int id = Interlocked.Increment(ref this.nextId);
-            return InvokeCoreAsync<Result>(id, targetName, arguments, cancellationToken);
+            return InvokeWithCancellationCoreAsync<Result>(targetName, arguments, cancellationToken, isParameterObject: false);
```

Comment:
I'm a little concerned that derived types that override the original `InvokeCoreAsync<T>` method will find that in code that doesn't leverage your new functionality their override is no longer invoked. Can we rearrange these thunks so that the param array style still calls through the original virtual method and only the new param object style goes directly to the new one?  Our own implementation of the old virtual method can still simply forward the call to the new method, but this way an override could still take effect.